api/graphite.go

```diff
@@ -568,7 +568,27 @@ func (s *Server) executePlan(ctx context.Context, orgId int, plan expr.Plan) ([]
 	// e.g. target=movingAvg(foo.*, "1h")&target=foo.*
 	// note that in this case we fetch foo.* twice. can be optimized later
 	for _, r := range plan.Reqs {
-		series, err := s.findSeries(ctx, orgId, []string{r.Query}, int64(r.From))
+		var err error
+		var series []Series
+		if strings.HasPrefix(r.Query, "seriesByTag(") {
+			startPos := strings.Index(r.Query, "(")
+			endPos := strings.LastIndex(r.Query, ")")
+			exprs := strings.Split(r.Query[startPos+1:endPos], ",")
+			log.Info("Resolving seriesByTarget exprs = %v", exprs)
+			series, err = s.clusterFindByTag(ctx, orgId, exprs, int64(r.From))
+			// if len(series) > 0 {
```

Comment:
For something like `sumSeries(seriesByTag('name=cpu.percent.idle'))` graphite replaces the seriesByTag expression with the expanded names as `sumSeries(cpu.percent.idle;host=h1,cpu.percent.idle;host=h2,cpu.percent.idle;host=h3,...,cpu.percent.idle;host=hN)`

Currently, I leave it as `sumSeries(seriesByTag('name=cpu.percent.idle'))`, since it didn't work when I changed the value of `r.Query` (probably caused a cache miss?)

Reply:
I didn't verify that now by reading the code again, but if i recall correctly graphite works like this that it first submits all the requests to the backends, then collects the returned results, and then uses the results based on the `target` string that's defined inside the result. That means if a backend returns result with a different `target`, then graphite will simply not use that one and consider the target it has been looking for to be empty.

Reply:
In MT. Changing `r.Query` caused no matches.